src/basic/REDUCE3_INT-Sycl.cpp

```diff
@@ -0,0 +1,133 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2017-24, Lawrence Livermore National Security, LLC
+// and RAJA Performance Suite project contributors.
+// See the RAJAPerf/LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#include "REDUCE3_INT.hpp"
+
+#include "RAJA/RAJA.hpp"
+
+#if defined(RAJA_ENABLE_SYCL)
+
+#include "common/SyclDataUtils.hpp"
+
+#include <iostream>
+
+namespace rajaperf
+{
+namespace basic
+{
+
+#define REDUCE3_INT_DATA_SETUP_SYCL \
+  Int_ptr hsum; \
+  allocAndInitSyclDeviceData(hsum, &m_vsum_init, 1, qu); \
+  Int_ptr hmin; \
+  allocAndInitSyclDeviceData(hmin, &m_vmin_init, 1, qu); \
+  Int_ptr hmax; \
+  allocAndInitSyclDeviceData(hmax, &m_vmax_init, 1, qu);
+
+#define REDUCE3_INT_DATA_TEARDOWN_SYCL \
+  deallocSyclDeviceData(hsum, qu); \
+  deallocSyclDeviceData(hmin, qu); \
+  deallocSyclDeviceData(hmax, qu);
+
+template <size_t work_group_size >
+void REDUCE3_INT::runSyclVariantImpl(VariantID vid)
+{
+  const Index_type run_reps = getRunReps();
+  const Index_type ibegin = 0;
+  const Index_type iend = getActualProblemSize();
+
+  REDUCE3_INT_DATA_SETUP;
+
+  if ( vid == Base_SYCL ) {
+
+    REDUCE3_INT_DATA_SETUP_SYCL;
+
+    startTimer();
+    for (RepIndex_type irep = 0; irep < run_reps; ++irep) {
+
+      const size_t global_size = work_group_size * RAJA_DIVIDE_CEILING_INT(iend, work_group_size);
+
+      initSyclDeviceData(hsum, &m_vsum_init, 1, qu);
+      initSyclDeviceData(hmin, &m_vmin_init, 1, qu);
+      initSyclDeviceData(hmax, &m_vmax_init, 1, qu);
```

Comment:
Combine into a single buffer?

Reply:
Yes!  I was going to deal with that when I convert the SYCL kernels with reductions to the new RAJA reduction API.